src/main/java/org/littleshoot/proxy/impl/UpstreamConnectionHandler.java

```diff
@@ -0,0 +1,41 @@
+package org.littleshoot.proxy.impl;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponse;
+
+public class UpstreamConnectionHandler extends SimpleChannelInboundHandler<Object> {
```

Comment:
this handler uses `SimpleChannelInboundHandler ` which automatically releases the msg. I am still testing that